app/src/main/java/com/example/atheneum/activities/AddBookActivity.java

```diff
@@ -0,0 +1,231 @@
+/*
+ * Copyright <YEAR> <COPYRIGHT HOLDER>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package com.example.atheneum.activities;
+
+import android.app.Activity;
+import android.content.Context;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.design.widget.FloatingActionButton;
+import android.support.design.widget.Snackbar;
+import android.support.v7.app.AppCompatActivity;
+import android.support.v7.widget.Toolbar;
+import android.text.TextUtils;
+import android.util.Log;
+import android.view.View;
+import android.view.inputmethod.InputMethodManager;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.Toast;
+
+import com.example.atheneum.R;
+import com.example.atheneum.fragments.AddBookFragment;
+import com.example.atheneum.fragments.OwnerPageFragment;
+import com.example.atheneum.models.Book;
+import com.example.atheneum.models.User;
+import com.example.atheneum.utils.EditTextWithValidator;
+import com.example.atheneum.utils.NonEmptyTextValidator;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseUser;
+import com.google.firebase.database.DataSnapshot;
+import com.google.firebase.database.DatabaseError;
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.FirebaseDatabase;
+import com.google.firebase.database.ValueEventListener;
+
+public class AddBookActivity extends AppCompatActivity {
```

Comment:
What was the purpose of recreating your entire AddNewBookFragment here? You could just inject the fragment here via xml if you want to avoid having the navbar.

Reply:
But if you want to completely remove the AddNewBook I would understand but since it was already created you could've just removed it from the MainActivity and had it here.

Reply:
This was done so that we could delete it after the merge, since I was working on stuff in the fragment version(the API call things) and I was going to move that to the activity and delete the fragment in my own PR